pkg/build/builder/docker.go

```diff
@@ -413,6 +419,33 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV Dockerfile instruction right after the
+// FROM instruction of the given dockerfile to define the environment variables
+// from env and returns the new Dockerfile.
+func insertEnvAfterFrom(env []kapi.EnvVar, dockerfile []byte) []byte {
+	if len(env) == 0 {
+		// Just return a copy.
+		return append([]byte(nil), dockerfile...)
```

Comment:
> and why not check for this in the place we call insertEnvAfterFrom and just skip calling that function of there are no envs to insert?

@mfojtik, by itself the function should behave correctly. If you remove the check, `insertEnvAfterFrom([]kapi.EnvVar{}, dockerfile)` will return something like this:

```
FROM centos
ENV

LABEL ...

RUN ...
```

Operating on the different forms the input can take is a natural part of a function definition.